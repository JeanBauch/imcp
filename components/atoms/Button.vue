<template>
	<div>
		<button v-if="mediumSize" type="button" :name="title" :class="themeColors.background" class="flex items-center justify-center px-6 py-2 md:px-9 md:py-3 rounded-full">
			<h4 class="text-white font-medium text-base md:text-lg">
				{{ title.toUpperCase() }}
			</h4>
		</button>
		<button v-else type="button" :name="title" :class="themeColors.background" class="flex items-center justify-center px-4 py-2 rounded-full">
			<h5 class="text-white font-medium text-base">
				{{ title.toUpperCase() }}
			</h5>
		</button>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	data () {
		return {
			themeColors: {
				background: `bg-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-500`,
				backgroundOnHover: `hover:bg-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-500`,
			},
		};
	},
	props: {
		title: { type: String, required: true },
		mediumSize: {
			type: Boolean,
			default: true,
		},
		color: {
			required: true,
			type: String,
			default: 'yellow',
			validator: (value: string) => [
				'yellow',
				'orange',
				'bordo',
				'beige',
				'blue',
			].includes(value.toLowerCase()),
		},
	},
});
</script>
