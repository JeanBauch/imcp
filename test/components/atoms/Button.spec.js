import { shallowMount } from '@vue/test-utils';

import Button from '~/components/atoms/Button.vue';

describe('Components:atoms:Button', () => {
	const factory = (values = {}) => {
		return shallowMount(Button, {
			propsData: {
				...values,
			},
		});
	};
	describe('Snapshots', () => {
		it('should render component', () => {
			const wrapper = factory({ title: 'Saiba Mais', color: 'blue' });
			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render correctly data as with props datas', () => {
			const wrapper = factory({ title: 'Saiba Mais', color: 'blue' });

			expect(wrapper.find('.bg-secondary-blue-500').exists()).toBeTruthy();
		});

		it('should render correctly data with color:blue', async () => {
			const wrapper = factory({ title: 'Saiba Mais', color: 'blue' });

			await wrapper.setData({ themeColors: { background: 'bg-secondary-blue-500' } });
			await wrapper.setData({ themeColors: { backgroundOnHover: 'hover:bg-secondary-blue-500' } });

			expect(wrapper.vm.themeColors.background).toBe('bg-secondary-blue-500');
			expect(wrapper.vm.themeColors.backgroundOnHover).toBe('hover:bg-secondary-blue-500');
		});

		it('should render correctly data with color:yellow', async () => {
			const wrapper = factory({ title: 'Saiba Mais', color: 'yellow' });

			await wrapper.setData({ themeColors: { background: 'bg-primary-yellow-500' } });
			await wrapper.setData({ themeColors: { backgroundOnHover: 'hover:bg-primary-yellow-500' } });

			expect(wrapper.vm.themeColors.background).toBe('bg-primary-yellow-500');
			expect(wrapper.vm.themeColors.backgroundOnHover).toBe('hover:bg-primary-yellow-500');
		});
	});
});
