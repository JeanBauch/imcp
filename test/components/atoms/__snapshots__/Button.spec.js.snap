// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:atoms:Button Snapshots should render component 1`] = `
"<div><button type=\\"button\\" name=\\"Saiba Mais\\" class=\\"flex items-center justify-center px-6 py-2 md:px-9 md:py-3 rounded-full bg-secondary-blue-500\\">
    <h4 class=\\"text-white font-medium text-base md:text-lg\\">
      SAIBA MAIS
    </h4>
  </button></div>"
`;
